v2/blockstore/readwrite.go

```diff
@@ -284,22 +286,22 @@ func (b *ReadWrite) unfinalize() error {
 	return err
 }
 
-func (b *ReadWrite) panicIfFinalized() {
-	if b.header.DataSize != 0 {
-		panic("must not use a read-write blockstore after finalizing")
-	}
+func (b *ReadWrite) finalized() bool {
+	return b.header.DataSize != 0
```

Comment:
Is it safe to access this value outside of a lock?

Reply:
Finalize is the only one writing to it. So it's only unsafe if one calls Finalize twice concurrently. Calling Finalize twice already errors, and the method is not documented as safe for concurrent use.